### ISSUE TITLE
Training on datasets from the paper (NIST)

### ISSUE THREAD

**Issue by seth814** [pengzhou1108/RGB-N#12]
Hey peng,

I'm been able to train a model using the synthetic dataset you provided. I would like to use this pretrained model and train further using NIST2016. I believe this is the "Use synthetic pre-trained model for fine tuning" step. I'm using the NC2016_Test0601 folder which was provided in the download of NIS2016. I'm assuming we have to create the txt files similar to how they are in the provided synthetic dataset? Since it says I am missing them... 

I can generate these files easily enough using the ground truth masks, but I'm confused about the format. For example...

In train_filter_single.txt you have:

Tp_75466_471421_31.84_62.29_640.0_251.24_airplane.png
bbox: [12, 66, 640, 368] (red)

![image](https://user-images.githubusercontent.com/17873279/53436647-ac186380-39c9-11e9-8c3e-033978f430e8.png)

Based on the code in split_train_test.py, it seems train_filter.txt is created by cropping the original image from coco and rescaling it to the aspect ratio of the new image. So this blue roi is padded to create the green bbox. Then the green bbox is rescaled to create the red one from above?

![image](https://user-images.githubusercontent.com/17873279/53437049-9f483f80-39ca-11e9-8be8-80b445983e7c.png)

So I'm asking, which files are required to train using a dataset like NIST? Just the tampered images found in train_filter_single.txt? I'm assuming I would just use the masks and create bboxes for each image from NIST.

**Comment by seth814** [comment]
An update of this:

I prepared the Columbia dataset: http://www.ee.columbia.edu/ln/dvmm/downloads/authsplcuncmp

I used the masks to create a bounding box. Put all the images in a data directory within RGB-N. Generated a txt file called train_columbia_single.txt. Formatted exactly like the provided synthetic coco download. For example:

columbia_file_1.tif 0 47 720 528 tamper
columbia_file_2.tif 23 231 528  567 tamper
...

I modified --weights and variables to use NIST as the ${DATASET} and NIST_columbia_bilinear_new as ${NET}

Training will reference lib/datasets
So change nist_path in factory.py to the location of your prepared dataset ( /home/ubuntu/RGB-N/data/columbia ) for me. Because I used nist it will load using lib/datasets/nist.py.

After training on Columbia using the model I trained from coco synthetic, I got some results after 8k iterations. Here are some examples after running a modification of demo.py:

![image](https://user-images.githubusercontent.com/17873279/53605741-78366d00-3b86-11e9-87b5-287b500e77b2.png)

![image](https://user-images.githubusercontent.com/17873279/53605854-ba5fae80-3b86-11e9-9865-7b2ad7dd4a95.png)

**Comment by 28150722** [comment]
> An update on this:
> 
> I prepared the Columbia dataset: http://www.ee.columbia.edu/ln/dvmm/downloads/authsplcuncmp
> 
> I used the masks to create a bounding box. Put all the images in a data directory within RGB-N. Generated a txt file called train_columbia_single.txt. Formatted exactly like the provided synthetic coco download. For example:
> 
> columbia_file_1.tif 0 47 720 528 tamper
> columbia_file_2.tif 23 231 528 567 tamper
> ...
> 
> Modified train_faster_rcnn.sh:
> 
> --weights: 'path_to_coco_synthetic_model.ckpt'
> ${DATASET} : NIST
> ${NET} : NIST_columbia_bilinear_new
> 
> Training will reference lib/datasets
> So change nist_path in factory.py to the location of your prepared dataset ( /home/ubuntu/RGB-N/data/columbia ) for me. Because I used nist it will load using lib/datasets/nist.py.
> 
> After training on Columbia using the model I trained from coco synthetic, I got some results after 8k iterations. Here are some examples after running a modification of demo.py:
> 
> ![image](https://user-images.githubusercontent.com/17873279/53605741-78366d00-3b86-11e9-87b5-287b500e77b2.png)
> 
> ![image](https://user-images.githubusercontent.com/17873279/53605854-ba5fae80-3b86-11e9-9865-7b2ad7dd4a95.png)
> 
> I plan on using this dataset preparation to train on NIST2017 in the future.

Hello, I am trying to download the NIST2016 dataset from the Internet, but I did not find the download address on the Internet. Can you share the dataset download link you mentioned?

Other than that, I am a newbie researcher. Would you like to post code about generating boundbox?

Thank you！

**Comment by Israrulhaq190** [comment]
> An update on this:
> 
> I prepared the Columbia dataset: http://www.ee.columbia.edu/ln/dvmm/downloads/authsplcuncmp
> 
> I used the masks to create a bounding box. Put all the images in a data directory within RGB-N. Generated a txt file called train_columbia_single.txt. Formatted exactly like the provided synthetic coco download. For example:
> 
> columbia_file_1.tif 0 47 720 528 tamper
> columbia_file_2.tif 23 231 528 567 tamper
> ...
> 
> Modified train_faster_rcnn.sh:
> 
> --weights: 'path_to_coco_synthetic_model.ckpt'
> ${DATASET} : NIST
> ${NET} : NIST_columbia_bilinear_new
> 
> Training will reference lib/datasets
> So change nist_path in factory.py to the location of your prepared dataset ( /home/ubuntu/RGB-N/data/columbia ) for me. Because I used nist it will load using lib/datasets/nist.py.
> 
> After training on Columbia using the model I trained from coco synthetic, I got some results after 8k iterations. Here are some examples after running a modification of demo.py:
> 
> ![image](https://user-images.githubusercontent.com/17873279/53605741-78366d00-3b86-11e9-87b5-287b500e77b2.png)
> 
> ![image](https://user-images.githubusercontent.com/17873279/53605854-ba5fae80-3b86-11e9-9865-7b2ad7dd4a95.png)
> 
> I plan on using this dataset preparation to train on NIST2017 in the future.

@seth814 

Could you please share the generated dataset for columbia in the form of text file(for image information and bounding box with class) as you explained above.

Thanks
